chia/server/server.py

```diff
@@ -141,6 +141,7 @@ class ChiaServer:
     received_message_callback: Optional[ConnectionCallback] = None
     banned_peers: Dict[str, float] = field(default_factory=dict)
     invalid_protocol_ban_seconds = INVALID_PROTOCOL_BAN_SECONDS
+    allow_banning_localhost: bool = False
```

Comment:
Do we have tests for the case when this is false, that we aren't allowed to ban? Seems like if this were to get messed up in a future change, banning localhost could have a lot of potentially bad consequences for services that need to talk to each other locally to keep a farm running